File: 프로그래머스/unrated/133501. 야간 전술보행/야간 전술보행.py
def solution(distance, scope, times):
    answer = [distance]

    for sc_index, sc in enumerate(scope):
        a = sum(times[sc_index])
        b = times[sc_index][0]
        for s in range(min(sc), max(sc)+1):
            if 0 < s % a <= b:
                answer.append(s)

    return min(answer)

File: 프로그래머스/unrated/133501. 야간 전술보행/야간 전술보행.js
function solution(distance, scopes, times) {
    scopes = scopes.map(scope => scope.sort((a,b)=> a-b))
    console.log(scopes)
    res = distance
    scopes.forEach(([s,e],i) => {
        const [w,r] = times[i]
        for (let d = s; d<=e; d++) {
            if (0<d%(w+r) &&  d%(w+r) <= w) {
                res = Math.min(res,d)
            }
        }
    })
    return res
}